File: src/main/java/com/jhl/mds/dto/MySQLFieldWithMappingDTO.java
package com.jhl.mds.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MySQLFieldWithMappingDTO {
    private String sourceField;
    private String targetField;
    private boolean mappable;
}

File: src/main/resources/static/resources/js/source/dto/my-sqlfield-with-mapping-dto.js
// @flow
import {observable} from 'mobx';

export class MySQLFieldWithMappingDTO {

    /**
     * @type {string}
     */
    @observable sourceField: ?string = '';
    /**
     * @type {string}
     */
    @observable targetField: ?string = '';
    /**
     * @type {boolean}
     */
    @observable mappable: ?boolean = false;

    /**
     * @param sourceField {string}
     * @param targetField {string}
     * @param mappable {boolean}
     */
    constructor(sourceField: ?string, targetField: ?string, mappable: ?boolean) {
        this.sourceField = sourceField;
        this.targetField = targetField;
        this.mappable = mappable;
    }


}
